### DISCUSSION TITLE
react-router migration: history.subscribe() in TanStack

### DISCUSSION THREAD

**Discussion by karuifeather** [TanStack/router#898] [Q&A]
Hey, I was just trying out TanStack's router as an alternative to react-router due to the breaking changes from v5 to v6 that broke my app.

My question is how do I achieve something like this with the new router. 

``` js
import {createMemoryHistory} from 'history'

const history = createMemoryHistory();

history.listen(({ action, location }) => {
  // The current location changed.
});
```

I noticed, unlike with the history object from the `history` library, callback to `history.subscribe()` does not get passed any parameters. Maybe I'm missing something very basic but I just want to pass the `location` object back to the callback function.

```js
import { createMemoryHistory } from '@tanstack/react-router';
const history = createMemoryHistory();
history.subscribe(onNavigate);
```
`onNavigate` is the callback function.

**Comment by tannerlinsley** [comment]
Correct, there is no action (yet), but you can just do this:

const unsubFn = history.subscribe(() => {
  myCalback(history.location)
})

**Reply by karuifeather** [reply]
Yep, ended up doing just that. I know it's late reply but thanks!
